setup.py

```diff
@@ -2,18 +2,22 @@
 import os
 import sys
 
+from packaging.version import parse
 from setuptools import find_packages, setup
 
-min_version, max_version = ((3, 9), "3.9"), ((3, 11), "3.11")
-if not (min_version[0] <= sys.version_info[:2] <= max_version[0]):
-    # Python 3.5 does not support f-strings
+supported_python_versions = ["3.9", "3.10", "3.11"]
+
+python_versions = [parse(v) for v in supported_python_versions]
+min_version = min(python_versions)
+max_version = max(python_versions)
+if not (
+    min_version <= parse(".".join([str(v) for v in sys.version_info[:2]])) <= max_version
+):
     py_version = ".".join([str(v) for v in sys.version_info[:3]])
     error = (
         "\n----------------------------------------\n"
-        "Error: This repo requires python {min_version}-{max_version}.\n"
-        "You are running python {py_version}".format(
-            min_version=min_version[1], max_version=max_version[1], py_version=py_version
-        )
+        f"Error: This repo requires python {min_version.base_version}-{max_version.base_version}.\n"
+        f"You are running python {py_version}."
```

Comment:
Nit: f-strings are only supported starting in 3.5(?). If you use them here and someone is using an older python they won't get the appropriate message telling them what versions are supported.

Reply:
Oh yeah, that's not useful. Will put back